Lib/test/test_unicode.py

```diff
@@ -625,13 +625,21 @@ def test_isspace(self):
 
     @support.requires_resource('cpu')
     def test_isspace_invariant(self):
+        '''
+        Before Python 3.9, `str.isspace` used a certain formula
+        combining the General Category and the Bidirectional Class.
+
+        Since Python 3.9, we use the White_Space property that Unicode
+        now provides.  Check that these differ only on U+001C..U+001F .
+        '''
         for codepoint in range(sys.maxunicode + 1):
             char = chr(codepoint)
             bidirectional = unicodedata.bidirectional(char)
             category = unicodedata.category(char)
             self.assertEqual(char.isspace(),
-                             (bidirectional in ('WS', 'B', 'S')
-                              or category == 'Zs'))
+                             ((bidirectional in ('WS', 'B', 'S')
+                               or category == 'Zs')
+                              and codepoint not in range(0x1c, 0x20)))
```

Comment:
Is this guaranteed to hold in future unicode versions? i.e. should it be tested on the string method if it's only testing the unicode database parsing and generation?